ISSUE: vcmap does not work with latest tbb (2021.1)

<!--
Before creating new issue, ensure that similar issue wasn't already created
  * Search: https://github.com/pmem/pmemkv/issues

Note that if you do not provide enough information to reproduce the issue, we may not be able to take action on your report.
Remember this is just a minimal template. You can extend it with data you think may be useful.
-->

# ISSUE: vcmap does not work with latest tbb (2021.1)

## Environment Information

- pmemkv version(s):                                                 1.0/1.1/1.2/1.3/master
- libpmemobj-cpp version(s):                                         <!-- fill this out -->
- PMDK (libpmem/libpmemobj) package version(s):                      <!-- fill this out -->
- OS(es) version(s):                                                 <!-- fill this out -->
- kernel version(s):                                                 <!-- fill this out -->
- compiler, libraries, packaging and other related tools version(s): <!-- fill this out -->

and possibly:
- TBB version(s):                                                   2021.1
- memkind version(s):                                                <!-- fill this out -->
- ndctl version(s):                                                  <!-- fill this out -->
<!-- fill in also other useful environment data -->

## Please provide a reproduction of the bug:

```
/home/igor/installed/tbb-2021-1.1-lock/include/tbb/../oneapi/tbb/concurrent_hash_map.h:785:31: error: call to implicitly-deleted default constructor of 'tbb::detail::d1::tbb_hash_compare<std::__cxx11::basic_string<char, std::char_traits<char>, libmemkind::pmem::allocator<char> > >'
        : concurrent_hash_map(HashCompare(), a)

```

## How often bug is revealed:

always

## Actual behavior:

Compilation fails

## Expected behavior:

Compilation passes

## Details

It looks like tbb specialization for std::hash<std::basic_string> is not found. Maybe this has something to do with our build system - standalone program with tbb::concurrent_hash_map and std::basic_string with memkind allocator works fine.

After fixing this, we should add a CI build with latest tbb.

## Additional information about Priority and Help Requested:

Are you willing to submit a pull request with a proposed change? (Yes, No)  <!-- check one if possible -->

Requested priority: (Showstopper, High, Medium, Low)                        <!-- check one if possible -->

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -215,6 +215,7 @@ endif()
 
 if(ENGINE_VCMAP)
 	include(tbb)
+	add_definitions(-DTBB_DEFINE_STD_HASH_SPECIALIZATIONS)
 	list(APPEND PKG_CONFIG_REQUIRES tbb)
 	list(APPEND RPM_DEPENDS tbb)
 	list(APPEND DEB_DEPENDS libtbb2)

